app/js/enhancedDate.js

```diff
@@ -1,5 +1,107 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iffe() {
+  'use strict';
 
+  //variables for all of the functions below declared here
+
+
+  var myDate;
+  var myDayName;
+  var dayNames = ["Sunday", "Monday","Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+  var myMonth;
+  var monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+  var todaysDate;
+
+
+  return {
+
+
+    setDate: function setDate(enterDate) {
+        if (enterDate instanceof Date){
+          myDate = enterDate;
```

Comment:
This works, but just keep in mind that objects are passed by reference when used as arguments in JavaScript. What this means is that if the object that was passed in was modified outside the module, it would modify your internal reference to it as well. Instead, you can just instantiate a new Date object by passing in the value passed in to setDate to the Date constructor: `myDate = new Date(enterDate);` This would then automatically handle the other value a user can pass in, milliseconds. So a check like this: `if (enterDate instanceof Date || typeof enterDate === 'number')` and you would then just pass in enterDate to the Date constructor, otherwise you'd do the `new Date()` without passing in anything (as you've done.) For the millisecond number, you want to create a Date object out of it so you don't need to convert it to a Date object later.